src/NeoSharp.Core/Network/Tcp/TcpPeer.cs

```diff
@@ -125,7 +125,8 @@ public bool ChangeProtocol(VersionPayload version)
         public void Disconnect()
         {
             Dispose();
-            _logger.LogInformation("The peer was disconnected");
+            _logger.LogInformation($"The peer {this.EndPoint.Host}:{this.EndPoint.Port} was disconnected");
+            this._stream.Dispose();
```

Comment:
The stream has to be disposed already in Dispose method call.

Reply:
You are right!!! Still trying here... and learning a little bit about this.

the problem is this: after the peer is been disconnected I could not find a way to connected it again ... I try to dispose everything in order to call again Socket.Connect I got always a message saying: "the current socket is connected." (or something like this.

this will be clean up